pallets/ocex/src/lib.rs

```diff
@@ -98,18 +129,118 @@ decl_module! {
         /// It helps to notify enclave about sender's intend to withdraw via on-chain
         #[weight = 10000]
         pub fn withdraw(origin, asset_id:  AssetId, to: T::AccountId,amount: T::Balance) -> DispatchResult{
-            let sender: T::AccountId = ensure_signed(origin)?;
+            let _: T::AccountId = ensure_signed(origin)?;
             Self::deposit_event(RawEvent::TokenWithdrawn(asset_id, to, amount));
             Ok(())
         }
 
+        /// Register MainAccount
+        #[weight = 10000]
+        pub fn register(origin) -> DispatchResult{
+            let sender: T::AccountId = ensure_signed(origin)?;
+            ensure!(!<MainAccounts<T>>::contains_key(&sender), Error::<T>::AlreadyRegistered);
+            Self::register_acc(sender.clone())?;
+            Self::deposit_event(RawEvent::MainAccountRegistered(sender));
+            Ok(())
+        }
+
+        /// Add Proxy
+        #[weight = 10000]
+        pub fn add_proxy(origin, proxy: T::AccountId) -> DispatchResult{
```

Comment:
I think we also have to ensure that the given proxy is not added to some other main account. For that, we also have to maintain superset in storage i.e. ProxyAccount -> MainAccount

Reply:
As Alain described, when an order is placed the tx contains main account that should be used. Also this open the possibility of having the same proxy for multiple main accounts. 

Example,
A market maker using two main accounts for trading but uses the same proxy ( aka trading bot).

It's a feature.

Reply:
hmm ... got it